gossip/gossip.go

```diff
@@ -218,13 +201,22 @@ func (g *Gossip) GetNodeDescriptor(nodeID proto.NodeID) (proto.NodeDescriptor, e
 // GetNodeDescriptor and internally by getNodeIDAddressLocked.
 func (g *Gossip) getNodeDescriptorLocked(nodeID proto.NodeID) (*proto.NodeDescriptor, error) {
 	nodeIDKey := MakeNodeIDKey(nodeID)
+
+	// We can't use GetInfoProto here because that method grabs the lock.
```

Comment:
Yeah, I could, but then we'd need `getInfoLocked` as well. It's ugly either
way, and this particular use is going away when zones move to tables.
On Aug 26, 2015 6:19 PM, "Spencer Kimball" <notifications@github.com> wrote:

> In gossip/gossip.go
> <https://github.com/cockroachdb/cockroach/pull/2276#discussion_r38042320>:
>
> > @@ -218,13 +201,22 @@ func (g *Gossip) GetNodeDescriptor(nodeID proto.NodeID) (proto.NodeDescriptor, e
> >  // GetNodeDescriptor and internally by getNodeIDAddressLocked.
> >  func (g *Gossip) getNodeDescriptorLocked(nodeID proto.NodeID) (*proto.NodeDescriptor, error) {
> >  	nodeIDKey := MakeNodeIDKey(nodeID)
> > +
> > +	// We can't use GetInfoProto here because that method grabs the lock.
>
> Why not make a getInfoProtoLocked method and call that from GetInfoProto
> so you don't have to duplicate this unmarshaling code?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/2276/files#r38042320>.
>